catwalk/storage.py

```diff
@@ -218,12 +228,44 @@ def _get_head_of_matrix(self):
         except pandas.error.EmptyDataError:
             self._head_of_matrix = None
 
-
     def _load(self):
-        self._matrix = pandas.read_hdf(self.matrix_path, mode='r+')
-        with open(self.metadata_path) as f:
-            self._metadata = yaml.load(f)
-        self._matrix.set_index(self.metadata['indices'], inplace=True)
+        with smart_open.smart_open(self.matrix_path, "rb") as f:
+            self._matrix = self._read_hdf_from_buffer(f)
+        # with smart_open.smart_open(self.metadata_path, "rb") as f:
+        #     y = []
+        #     for line in f:
+        #         y.append(line.decode())
+        #self._metadata = yaml.load("".join(y).encode('utf-8'))
+        self._metadata = self.load_yaml(self.metadata_path)
+        try:
+            self._matrix.set_index(self._metadata['indices'], inplace=True)
+        except:
+            pass
+
+    def _read_hdf_from_buffer(self, buffer):
+        with pandas.HDFStore(
+                "data.h5",
+                mode="r",
+                driver="H5FD_CORE",
+                driver_core_backing_store=0,
+                driver_core_image=buffer.read()) as store:
+            return store[store.keys()[0]]
```

Comment:
yes, in theory there could be other keys because of the nature of hdf, but the whole framework shouldn't allow this to happen. Maybe I should raise an error when there's more than one key?